db/db_iter.h

```diff
@@ -417,6 +418,36 @@ class DBIter final : public Iterator {
     return true;
   }
 
+  void MarkMemtableForFlushIfNeeded() {
+    if (avg_op_scan_flush_trigger_ &&
+        mem_hidden_op_since_seek_ >= memtable_op_scan_flush_trigger_ &&
+        mem_hidden_op_since_seek_ >=
+            static_cast<uint64_t>(iter_step_since_seek_) *
+                avg_op_scan_flush_trigger_) {
```

Comment:
if we are worried about overflow, I think we should be able to afford a division in MarkMemtableForFlushIfNeeded() cuz it only happens in Seeks and destruction.